subsys/net/ip/ipv4_autoconf.c

```diff
@@ -0,0 +1,280 @@
+/** @file
+ * @brief IPv4 autoconf related functions
+ */
+
+/*
+ * Copyright (c) 2017 Matthias Boesl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
```

Comment:
I wonder if it is enough if we have
SPDX-License-Identifier: Apache-2.0
line here instead of full license text. Just pondering this thought here, no need to change this if you think you want to have full text here.